math/validate_email(filter).py

```diff
@@ -0,0 +1,69 @@
+import re
+
+
+def check_username(username):
+    valid_char = re.compile(r"[A-Za-z0-9-_]+")
+    if re.fullmatch(valid_char, username):
+        return True
+    else:
+        return False
+    
+
+def check_companyname(companyname):
+    valid_char =  re.compile(r"[A-Za-z0-9]+")
+    if re.fullmatch(valid_char, companyname):
+        return True
+    else:
+        return False
+
+def check_extension(extension):
+    valid_char = re.compile(r"[A-Za-z]+")
+    if re.fullmatch(valid_char, extension):
+        return True
+    else:
+        return False
+    
+
+def check_all_extension(extensions):
+    if len(extensions)>3:
+        return False
+    for extension in extensions:
```

Comment:
try use `all()`
`if not all(map(check_extension, extensions)`
or shorter
`return len(extensions) <=  3 and all(map(check_extension, extensions)`
but using `for` will reduce complexity.